### PR TITLE
Fixing SDL2 video backend implementation; fix Linux builds.

### PR DESCRIPTION
**Pull request by vladikcomper**
This PR fixes initialization of texture for image display (use "SDL_TEXTUREACCESS_STREAMING" flag instead of "SDL_TEXTUREACCESS_STATIC" hint) and incorrect use of "SDL_GetWindowSurface" with the hardware-accelerated renderer.

It effectively fixes emulator crashes on Linux systems (and possibly, other platforms), though some extra testing may be required to retain compatibility with Windows and changing MD video modes by the game itself.

### CODE REVIEW

**Comment by vladikcomper** [comment]
This PR was extended with a fix for framerate timing issues.

File: src/backends/video/video_sdl2.cpp [hunk 1]
```diff
@@ -170,8 +177,7 @@ void Update_Viewport() {
     rect_dest.y *= 0.9343065693430657;
   #endif
 
-  /* clear destination surface */
-  SDL_FillRect(sdl_winsurf, 0, 0);
+  // TODO: Clear window contents without relying on "SDL_GetWindowSurface".
```
Comment (heyjoeway) [inline src/backends/video/video_sdl2.cpp:180]: I think SDL_RenderClear will work for that. I'll handle that seperately.

**Comment by vladikcomper** [comment]
Most recent commits also fix issues with using SDL2mixer backend in Linux builds.

File: src/main.cpp [hunk 2]
```diff
@@ -396,28 +394,37 @@ int main (int argc, char *argv[]) {
...
 
   /* emulation loop */
   #ifdef __EMSCRIPTEN__
    emscripten_set_main_loop(&mainloop, 60, 1);
   #else
-    uint32 timeNext = 0;
-    // uint32 timePrev = 0;
+    timespec timeAfter, timeBefore;
+
     while(running) {
```
Comment (heyjoeway) [inline src/main.cpp:407]: Will add a way to use vsync for mainloop timing and fall back to this on refresh rates other than ~60Hz.

**Pull request merged**